### PR TITLE
fix(radio-button): radio name should be unique within its own scope

### PR DESCRIPTION
**Pull request by Theeraphat-Sorasetsakul**
## Description
[radio-button] radio group name should be unique within its own scope
If radio group is in document level, the name should be unique at document level.
If radio group is in shadow root level, the name should be unique at shadow root level.
jira: https://jira.refinitiv.com/browse/ELF-1754

Fixes # (issue)

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/elements/src/radio-button/__test__/radio-button.test.js [hunk 1]
```diff
@@ -447,6 +447,21 @@ describe('radio-button/RadioButton', () => {
...
+      radio2.setAttribute('name', 'group');
+      radio2.setAttribute('id', 'btn2');
+      radio2.checked = true;
+      container.shadowRoot.appendChild(radio2);
+      await elementUpdated(radio);
+      await elementUpdated(radio2);
+
+      expect(radio.checked).to.equal(true);
+      expect(radio2.checked).to.equal(true);
+    });
```
Comment (dtanp-rft) [inline packages/elements/src/radio-button/__test__/radio-button.test.js:487]: A better test case would be to create HTML like below and then check id=btn2 in both inside shadowroot as you did.

Expect that 
id=btn1 inside shadowroot => unchecked
id=btn2 inside  shadowroot => checked
id=btn1 outside  shadowroot => checked
id=btn2 outside  shadowroot => unchecked

Then we ensure that 1/ same group name of inside and outside shadowroot isn't interfere 2/ group is still working correctly if it's inside shadowroot.

```html
<ef-radio-button name="group" id="btn1" checked>1</ef-radio-button>
<ef-radio-button name="group" id="btn2">2</ef-radio-button>

<div>
   #shadowRoot
      <ef-radio-button name="group" id="btn1" checked>1</ef-radio-button>
     <ef-radio-button name="group" id="btn2">2</ef-radio-button>
</div>
```

**Review by dtanp-rft — CHANGES_REQUESTED** [review]

File: packages/elements/src/radio-button/__test__/radio-button.test.js (hunk 1 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/radio-button/__test__/radio-button.test.js:487]: Agree. I will do as your suggest.

**Review by Sarin-Udompanish — APPROVED** [review]

File: packages/elements/src/radio-button/__test__/radio-button.test.js [hunk 2]
```diff
@@ -447,6 +448,43 @@ describe('radio-button/RadioButton', () => {
       const checkedRadio = group.find(element => element.checked);
       expect(checkedRadio.id).to.equal('btn2');
     });
+    it('Should separate scope between shadow dom and light dom', async () => {
```
Comment (wsuwt) [inline packages/elements/src/radio-button/__test__/radio-button.test.js]:
```suggestion
    it('Should have separate registry when scope is between shadow DOM and light DOM', async () => {
```

Comment (dtanp-rft) [reply packages/elements/src/radio-button/__test__/radio-button.test.js]: I thought we're not going to split the registry? The registry is the same but when matching, we check the scope.

File: packages/elements/src/radio-button/radio-button-registry.ts [hunk 3]
```diff
@@ -93,7 +94,7 @@ const restoreTabIndex = (radioGroup: RadioButton[]): void => {
 };
 
 /**
- * Get the group of same name radio buttons
+ * Get the group of radio buttons that same name and scope
```
Comment (wsuwt) [inline packages/elements/src/radio-button/radio-button-registry.ts]:
```suggestion
 * Get a group of radio buttons that has the same name and scope
```

File: packages/elements/src/radio-button/__test__/radio-button.test.js (hunk 2 above)
Comment (wsuwt) [reply packages/elements/src/radio-button/__test__/radio-button.test.js]: Oops sorry.

Comment (wsuwt) [reply packages/elements/src/radio-button/__test__/radio-button.test.js]:
```suggestion
    it('Should separate scope between shadow DOM and light DOM', async () => {
```

**Review by Sakchai-Refinitiv — APPROVED** [review]

**Review by dtanp-rft — APPROVED** [review]

**Pull request merged**